lib/oauth.js

```diff
@@ -263,6 +263,9 @@ exports.v2 = function (settings) {
             if (typeof settings.clientSecret === 'string') {
                 query.client_secret = settings.clientSecret;
             }
+            else if (typeof settings.clientSecret === 'function') {
+                query.client_secret = settings.clientSecret();
```

Comment:
Is there any parameters that could be interesting to pass on here?

Reply:
`settings`?

Reply:
I'm not much of a bell user myself that's just a question that crossed my mind. I just wondered if that's something you considered on your end. If you don't need it let's keep it that way then.